shell/platform/darwin/ios/ios_surface.h

```diff
@@ -9,25 +9,37 @@
 
 #include <memory>
 
+#include "flutter/flow/embedded_views.h"
 #include "flutter/fml/macros.h"
 #include "flutter/fml/platform/darwin/scoped_nsobject.h"
 #include "flutter/shell/common/surface.h"
 
+@class CALayer;
+
 namespace flutter {
 
 // Returns true if the app explicitly specified to use the iOS view embedding
 // mechanism which is still in a release preview.
 bool IsIosEmbeddedViewsPreviewEnabled();
 
-class IOSSurface {
+class IOSSurface : public ExternalViewEmbedder {
  public:
-  IOSSurface(FlutterPlatformViewsController* platform_views_controller);
+  static std::unique_ptr<IOSSurface> Create(
+      std::shared_ptr<IOSContext> context,
+      fml::scoped_nsobject<CALayer> layer,
+      FlutterPlatformViewsController* platform_views_controller);
 
+  // |ExternalViewEmbedder|
   virtual ~IOSSurface();
 
-  virtual bool IsValid() const = 0;
+  std::shared_ptr<IOSContext> GetContext() const;
+
+  ExternalViewEmbedder* GetExternalViewEmbedderIfEnabled();
 
-  virtual bool ResourceContextMakeCurrent() = 0;
+  bool ResourceContextMakeCurrent();
+
+  // TODO(chinmaygarde): Required?
```

Comment:
This was a note to myself. The validity check is not used. This whole interface needs to be reworked. I am going to do that separately. Removed this comment.